lte/gateway/c/core/oai/tasks/nas/ies/ApnAggregateMaximumBitRate.c

```diff
@@ -112,53 +112,121 @@ int encode_apn_aggregate_maximum_bit_rate(
+
     apn_ambr->apnambrforuplink = 0xff;
-  } else if ((ambr_ul_kbps > 0) && (ambr_ul_kbps <= 63)) {
-    apn_ambr->apnambrforuplink = ambr_ul_kbps;
-  } else if ((ambr_ul_kbps > 63) && (ambr_ul_kbps <= 575)) {
-    apn_ambr->apnambrforuplink = ((ambr_ul_kbps - 64) / 8) + 64;
-  } else if ((ambr_ul_kbps > 575) && (ambr_ul_kbps <= 8640)) {
-    apn_ambr->apnambrforuplink = ((ambr_ul_kbps - 576) / 64) + 128;
-  } else if (ambr_ul_kbps > 8640) {
-    apn_ambr->apnambrforuplink = 0xfe;
-    apn_ambr->extensions =
+  }
+
+  if ((ambr_ul_kbps > 8640) &&
+      ((ambr_ul_kbps >= 8600) && (ambr_ul_kbps <= 16000))) {
+    apn_ambr->extensions |=
         APN_AGGREGATE_MAXIMUM_BIT_RATE_MAXIMUM_EXTENSION_PRESENT;
-    if ((ambr_ul_kbps >= 8600) && (ambr_ul_kbps <= 16000)) {
-      apn_ambr->apnambrforuplink_extended = (ambr_ul_kbps - 8600) / 100;
-    } else if ((ambr_ul_kbps > 16000) && (ambr_ul_kbps <= 128000)) {
-      apn_ambr->apnambrforuplink_extended =
-          ((ambr_ul_kbps - 16000) / 1000) + 74;
-    } else if ((ambr_ul_kbps > 128000) && (ambr_ul_kbps <= 256000)) {
-      apn_ambr->apnambrforuplink_extended =
-          ((ambr_ul_kbps - 128000) / 2000) + 186;
-    }
+    apn_ambr->apnambrforuplink_extended = (ambr_ul_kbps - 8600) / 100;
+    ambr_ul_kbps =
+        ambr_ul_kbps - (apn_ambr->apnambrforuplink_extended) * 100 - 8600;
+
+  } else if ((ambr_ul_kbps > 16000) && (ambr_ul_kbps <= 128000)) {
+    apn_ambr->extensions |=
+        APN_AGGREGATE_MAXIMUM_BIT_RATE_MAXIMUM_EXTENSION_PRESENT;
+    apn_ambr->apnambrforuplink_extended = ((ambr_ul_kbps - 16000) / 1000) + 74;
+    ambr_ul_kbps                        = ambr_ul_kbps -
+                   (apn_ambr->apnambrforuplink_extended - 74) * 1000 - 16000;
+
+  } else if ((ambr_ul_kbps > 128000) && (ambr_ul_kbps <= 256000)) {
+    apn_ambr->extensions |=
+        APN_AGGREGATE_MAXIMUM_BIT_RATE_MAXIMUM_EXTENSION_PRESENT;
+    apn_ambr->apnambrforuplink_extended =
+        ((ambr_ul_kbps - 128000) / 2000) + 186;
+
+    ambr_ul_kbps = ambr_ul_kbps -
+                   (apn_ambr->apnambrforuplink_extended - 186) * 2000 - 128000;
   }
 
-  apn_ambr->apnambrfordownlink_extended2 = 0;
-  apn_ambr->apnambrforuplink_extended2   = 0;
+  if ((apn_ambr->apnambrforuplink != 0xfe) && !(ambr_ul_kbps > 8640)) {
+    if (ambr_ul_kbps == 0) {
+      apn_ambr->apnambrforuplink = 0xff;
+    }
+    if ((ambr_ul_kbps > 0) && (ambr_ul_kbps <= 63)) {
+      apn_ambr->apnambrforuplink = ambr_ul_kbps;
+    } else if ((ambr_ul_kbps > 63) && (ambr_ul_kbps <= 575)) {
+      apn_ambr->apnambrforuplink = ((ambr_ul_kbps - 64) / 8) + 64;
+      ambr_ul_kbps =
+          ambr_ul_kbps - (apn_ambr->apnambrforuplink_extended - 64) * 8 - 63;
+    } else if ((ambr_ul_kbps > 575) && (ambr_ul_kbps <= 8640))
+
+    {
+      apn_ambr->apnambrforuplink = ((ambr_ul_kbps - 576) / 64) + 128;
+      ambr_ul_kbps =
+          ambr_ul_kbps - (apn_ambr->apnambrforuplink_extended - 128) * 64 - 575;
```

Comment:
apply linter to fix formatting. If you run `make build_oai`, it should automatically format it. Please before raising PRs, make sure that correctly formatted files are added/committed and pushed.

Reply:
Ran "make format_core" to format the code.